src/lib/mailer/logNotifications.js

```diff
@@ -0,0 +1,100 @@
+import { compileFile } from 'pug';
+import * as path from 'path';
+import { logger } from '../../logger';
+import { EMAIL_ACTIONS } from '../../constants';
+import { createMailerLog } from '../../services/mailerLog';
+
+const emailTemplatePath = path.join(process.cwd(), 'email_templates');
+
+/**
+ * Creates an email log entry for on-demand notifications
+ *
+ * @param {*} job - job containig data
+ * @param {boolean} success - success
+ * @param {*} result - notification result
+ *
+ */
+export default async function logEmailNotification(job, success, result) {
+  let subject; let emailTo; let
+    template;
+  let newCollaborator; let newApprover; let
+    collabArray;
+  let collaboratorEmailAddresses;
+  const { data } = job;
+  const { report } = data;
+  const { author, activityReportCollaborators } = report;
+
+  try {
+    switch (job.name) {
+      case EMAIL_ACTIONS.COLLABORATOR_ADDED:
+        newCollaborator = data.newCollaborator;
+        emailTo = newCollaborator ? [newCollaborator.email] : [''];
```

Comment:
I would assume we check the emailTo before trying to send out to blank recipient?